packages/waku/src/router/client.ts

```diff
@@ -522,9 +522,13 @@ export function Router({ routerData = DEFAULT_ROUTER_DATA }) {
                 window.location.pathname !== pathname ||
                 window.location.search.replace(/^\?/, '') !== searchParamsString
               ) {
-                routerData[1]?.forEach((listener) =>
-                  listener(pathname, searchParamsString),
-                );
+                setTimeout(() => {
+                  // HACK we need to wait a tick for React to start rendering.
+                  // FIXME It doesn't feel ideal. Hope to find a better solution.
+                  routerData[1]?.forEach((listener) =>
+                    listener(pathname, searchParamsString),
+                  );
+                });
```

Comment:
Would `startTransition` from react be less of a hack?

Reply:
Unfortunately, not. I'm not sure if it's React issue or Waku issue. Probably, the latter.

Meanwhile, I used another hack 3fedc84.

Reply:
Ahhh, unfortunately, this causes a flicker when moving to a new page. Let me revert it and open a new issue with `jump` case.

Reply:
#936

Reply:
#937